apps/sertra/components/ServerList.tsx

```diff
@@ -0,0 +1,87 @@
+import { useServerListings } from "../api/hooks";
+import { ServerListingData } from "../api/models";
+import { PropsWithChildren, useMemo } from "react";
+import styles from "./ServerList.module.css";
+
+interface ServerModeProps {
+  isPvP: boolean;
+}
+const ServerMode: React.FC<PropsWithChildren<ServerModeProps>> = ({
+  isPvP,
+}) => {
+  if (isPvP) {
+    return <div className={styles.pvp}>PvP</div>;
+  } else {
+    return <div className={styles.pve}>PvE</div>;
+  }
+};
+
+interface ServerPasswordProps {
+  requiresPassword: boolean;
+}
+const ServerPassword: React.FC<PropsWithChildren<ServerPasswordProps>> = ({
+  requiresPassword,
+}) => {
+  if (requiresPassword) {
+    return <div>🔒</div>;
+  } else {
+    return null;
+  }
+};
+
+interface ServerListEntryProps {
+  listing: ServerListingData;
+}
+
+const ServerListEntry: React.FC<PropsWithChildren<ServerListEntryProps>> = ({
+  listing,
+}) => {
+  return (
+    <div className={styles.row}>
+      <div className={styles.name}>{listing.name}</div>
+      <div className={styles.mode}>
+        <ServerMode isPvP={listing.is_pvp} />
+      </div>
+      <div className={styles.mods}>42</div>
+      <div className={styles.password}>
+        <ServerPassword requiresPassword={listing.requires_password} />
+      </div>
+    </div>
+  );
+};
+
+interface ServerListProps {
+  community: string;
+}
+export const ServerList: React.FC<PropsWithChildren<ServerListProps>> = ({
+  community,
+}) => {
+  const { data } = useServerListings();
+
+  // TODO: Pull from API
+  const gameDisplayName = useMemo(() => {
+    return community.replaceAll("-", " ");
+  }, [community]);
+
+  return (
+    <div className={styles.container}>
```

Comment:
@gkurck what's your take on divitis?

Reply:
Depends on the use case. I've had a case of divitis before when working with applications that are translated to native apps - or more specifically, wrapped in a webview inside a native app - because in this case SEO or accessibility issues aren't very glaring the same way.

That being said, for accessibility and SEO purposes semantic markup is generally preferred. This will help crawlers and screen readers to do their job better, although I'm not entirely sure how much weight crawlers put on semantics these days. I'd imagine they would. Semantic markup also plays an important role in keyboard navigation and maybe makes the source code quicker to read and understand.

Side note: `<ol>`stands for ordered list`, `<ul>` stands for `unordered list` - I find myself seldom having a need for `<ol>`.

Reply:
Backend returns the servers ordered by last modification date, so semantically we have an ordered list?

Reply:
Well, to split hairs, technically all lists are always in **some** order. `<ol>` usually gives the idea of a numbered list, e.g. 1., 2. 3. and so on with a definite meaning them being in that order. We will probably want to sort the list somehow in the future chaning the order without really changing the meaning, amirite?

Reply:
Maybe I'm splitting the hairs from a different perspective.

* From programmer's perspective, database guarantees order of the returned items only if ordering is explicitly requested. Otherwise the results will be in unspecified order (making the same request again may or may not change the order of returned items). In other words, they're `<ol>` and `<ul>`. **In this case**, we request the ordering explicitly.
* From a semantic perspective, I'd argue there **is** meaning in the order of the servers when they are returned from the backend. That meaning is changed if client shows "installed" servers at the top of the list, but there's still a meaning to the order. Likewise there is meaning in the order if user chooses to sort them by e.g. name. Sure, the meaning changes again, but there's meaning nevertheless. I guess the only way the ordering would become meaningless would be if we had "sort in random order" functionality.

The list above is in no specific order.

Reply:
You make good case and while what you said about backend is true, I must disagree with the semantics as HTML is backend agnostic, and instead semantically represents the structure of data to be presented for the end-user.

A good use case for ordered list would be the steps for a cooking recipe. While you can sort the steps by e.g. last modified, the meaning changes drastically as the step one of the recipe now tells you to garnish the omelet before breaking the eggs.

However in the case of the server list, the order of the items isn't definite, as the user can change the order as he or she pleases to reach a desired result of sorting. For example, we can have a list sorted by PVP-only, and then again by ascending or descending. Each sorting is as valid as the other, and doesn't invalidate the original order.

> The list above is in no specific order.

B-)